power_grid_model_c/power_grid_model_c/CMakeLists.txt

```
@@ -2,19 +2,64 @@
 #
 # SPDX-License-Identifier: MPL-2.0
 
-add_library(power_grid_model_c SHARED "power_grid_model_c.cpp")
+# The detailed headers
+add_library(power_grid_model_c_headers INTERFACE)
+
+target_include_directories(power_grid_model_c_headers INTERFACE
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/power_grid_model_c>
+  $<INSTALL_INTERFACE:include/power_grid_model_c>
```

now that i think about it, maybe this should be

```suggestion
  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
  $<INSTALL_INTERFACE:include>
```

the magic lives in the `PUBLIC_HEADER DESTINATION` below, I believe.